tests/integration/targets/ipam_range/tasks/main.yml

```diff
@@ -0,0 +1,297 @@
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].name == "Test Range"
+
+    - name: "Create a Range with exclusion ranges"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+        exclusion_ranges:
+          - start: "10.0.0.10"
+            end: "10.0.0.20"
+            comment: "Exclude this range"
+      register: range
+    - name: Get information about the Range
+      infoblox.bloxone.ipam_range_info:
+        filters:
+          start: "{{ range_start }}"
+          end: "{{ range_end }}"
+          space: "{{ _ip_space.id }}"
+      register: range_info
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].exclusion_ranges[0].start == "10.0.0.10"
+          - range_info.objects[0].exclusion_ranges[0].end == "10.0.0.20"
+          - range_info.objects[0].exclusion_ranges[0].comment == "Exclude this range"
+
+    - name: "Create a Range with threshold"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+        threshold:
+          enabled: true
+          high: 90
+          low: 10
+      register: range
+    - name: Get information about the Range
+      infoblox.bloxone.ipam_range_info:
+        filters:
+          start: "{{ range_start }}"
+          end: "{{ range_end }}"
+          space: "{{ _ip_space.id }}"
+      register: range_info
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].threshold.enabled == true
+          - range_info.objects[0].threshold.high == 90
+          - range_info.objects[0].threshold.low == 10
+
+    - name: "Create a Range with inheritance sources"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        inheritance_sources:
+          dhcp_options:
+            action: "block"
+        state: "present"
+      register: range
+    - name: Get information about the Range
+      infoblox.bloxone.ipam_range_info:
+        filters:
+          start: "{{ range_start }}"
+          end: "{{ range_end }}"
+          space: "{{ _ip_space.id }}"
+      register: range_info
+    - assert:
+        that:
+          - range is not failed
+          - range_info.objects | length == 1
+          - range_info.objects[0].inheritance_sources.dhcp_options.action == "block"
+
+
+  always:
+    - name: "Delete a Range"
+      infoblox.bloxone.ipam_range:
+        start: "{{ range_start }}"
+        end: "{{ range_end }}"
+        space: "{{ _ip_space.id }}"
+        state: "absent"
+
+    - ansible.builtin.include_role:
```

Comment:
Here I mean that there could be separate directories called cleanup_ip_space and cleanup_subnet which can be called in dependencies where the cleanup takes place.